components/CreateReplyRequestDialog/CreateReplyRequestDialog.js

```diff
@@ -60,25 +60,25 @@ class CreateReplyRequestDialog extends React.PureComponent {
     });
   }
 
-  set(key, value) {
-    this.setState({ [key]: value });
-
-    // Backup to localStorage
-    requestAnimationFrame(() => (localStorage[key] = value));
-  }
-
   handleTextChange = ({ target: { value } }) => {
     this.setState({
       text: value,
       disabled: !value || value.length < MIN_REASON_LENGTH,
     });
+
+    // Backup to localStorage
+    requestAnimationFrame(() => (localStorage.text = value));
   };
 
-  onReasonSubmitted = () => {
+  handleReasonSubmitted = () => {
     this.formRef.current.reset();
```

Comment:
Previously I suggested adding `reset()` because `<textarea>` was a [uncontrolled input](https://reactjs.org/docs/uncontrolled-components.html). However, in recent changes you provide `value` to `<textarea>`, which is turned into a [controlled input](https://reactjs.org/docs/forms.html#controlled-components).

I think we should take one of the following:

- If we want to make the `<textarea>` a controlled input, setting `this.state.text` to `''` can do the clearing. `reset()` and the `formRef` are no longer needed and should be removed.
- If we want to revert `<textarea>` back to an uncontrolled input, then change `value={text}` below to `defaultValue={text}`.

Reply:
I think this first depends on UX.
We make textarea controlled input because we want to save and reload the previous text.
So the question is whether we want to cache user's input or not.
This feature is implemented in ReplyForm, do users like that feature?

Reply:
Both controlled and uncontrolled input can implement automayic localStorage cache. The key difference between controlled and uncontrolled is `value` prop, not `onChange` prop.